src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs

```diff
@@ -517,8 +517,38 @@ internal void LoadPreferencesFileAtStartup()
         {
             if (viewModel.PreferenceSettings.IsFirstRun == true)
             {
-                //Move the current location two levels up
-                var programDataDir = Directory.GetParent(Directory.GetParent(viewModel.Model.PathManager.CommonDataDirectory).ToString()).ToString();
+                var programDataDir = string.Empty;
+
+                //This code will be executed only when Dynamo is running inside any Host application like Revit, FormIt, Civil3D
+                if (viewModel.Model.PathManager.HostApplicationDirectory != null && !string.IsNullOrEmpty(viewModel.Model.HostVersion))
```

Comment:
I would suggest adding a try-catch block here as we are interacting with directories.

Reply:
added the try/catch block in the next commit